docs/DeveloperGuide.md

```diff
@@ -1,18 +1,120 @@
 # Developer Guide
 
-## Design & implementation
+## Design
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
 
+The **Architecture Design** given above explains the high-level design of the App.
+Given below is a quick overview of each component. </br>
 
-## Product scope
-### Target user profile
+`Modman` is the main class of the application, and handles the app launch,
+initializing the appropriate classes to be used. </br>
+
+The rest of the app consists of four components:
+
+* Logic
+* UI
+* Storage
+* Data
+
+**How the four architecture components interact with each other and the ModMan class**
+
+The *Sequence diagram* below shows the interaction between components for when the user issues the command `add module CS2113T`
+
+![](uml/ArchitectureSequenceDiagram.png)
+
+The sections below give more details of each component.
+
+
+## Logic Component (Jianning)
+
+![logic](uml/ParserAndCommand.png)
+
+1. The `Parser` class is used to parse the user command.
+2. This returns a `Command` object which is executed in `Modman`.
+3. The command execution can affect the state of other components e.g. the sort command changes the order of assignments in `Module`.
+4. The `Command` object can also instruct the `Ui` to list and display information to the user.
+
+Given below is the Sequence Diagram for creating the corresponding `Command` object from user input via `Parser`. </br>
+The sequence diagram also acts as a reference frame for `getCommand`.
```

Comment:
It could be slightly confusing to a reader as to where /in what context getCommand is actually used, since getCommand is not referenced anywhere else. Perhaps it would be useful to include a brief definition of it?